sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -366,14 +375,138 @@ CosmosAsyncContainer setId(String id) {
     }
 
     /**
-     * Gets a CosmosAsyncItem object without making a service call
+     * Reads an item.
+     * <p>
+     * After subscription the operation will be performed.
+     * The {@link Mono} upon successful completion will contain a cosmos item response with the read item
+     * In case of failure the {@link Mono} will error.
+     *
+     * @param itemId the item id
+     * @param partitionKey the partition key
+     * @return an {@link Mono} containing the cosmos item response with the read item or an error
+     */
+    public <T> Mono<CosmosAsyncItemResponse<T>>  readItem(String itemId, PartitionKey partitionKey, Class<T> itemType) {
+        return readItem(itemId, partitionKey, new CosmosItemRequestOptions(partitionKey), itemType);
+    }
+
+    /**
+     * Reads an item.
+     * <p>
+     * After subscription the operation will be performed.
+     * The {@link Mono} upon successful completion will contain a cosmos item response with the read item
+     * In case of failure the {@link Mono} will error.
      *
-     * @param id id of the item
+     * @param itemId the item id
      * @param partitionKey the partition key
-     * @return a cosmos item
+     * @param options the request cosmosItemRequestOptions
+     * @return an {@link Mono} containing the cosmos item response with the read item or an error
      */
-    public CosmosAsyncItem getItem(String id, Object partitionKey) {
-        return new CosmosAsyncItem(id, partitionKey, this);
+    public <T> Mono<CosmosAsyncItemResponse<T>>  readItem(String itemId, PartitionKey partitionKey, 
+                                                  CosmosItemRequestOptions options, Class<T> itemType) {
+        if (options == null) {
+            options = new CosmosItemRequestOptions();
+        }
+        options.setPartitionKey(partitionKey);
+        RequestOptions requestOptions = options.toRequestOptions();
+        return this.getDatabase().getDocClientWrapper()
+                   .readDocument(getItemLink(itemId), requestOptions)
+                   .map(response -> new CosmosAsyncItemResponse<T>(response, itemType))
+                   .single();
```

Comment:
For V4, I think we can change our `AsyncDocumentClient` APIs to return Mono for single returns so that we won't have to call `.single()` APIs here. 
Right now they return Flux<> for readItem which doesn't make sense. 
I believe @moderakh raised this point with v3. 
We should correct that behavior for v4 for all the APIs if they return a single document / item. 

Calling this extra `.single()` when not needed will affect the performance of these APIs.

Reply:
I don't think this specific case may have any impact on perf as it has to be done somewhere. you are not removing the need for invoking `single()` you are moving it from here to `AsyncDocumentClient`.

and yes ideally it has to be in `AsyncDocumentClient` for cleanness although not directly related to this PR. IMO can be done separately as a good improvement in a different PR.

Reply:
+1. We can do it in a separate PR if necessary.